src/Symfony/Component/Cache/Adapter/ProxyAdapter.php

```diff
@@ -43,32 +44,58 @@ public function __construct(CacheItemPoolInterface $pool, string $namespace = ''
             function ($key, $innerItem) use ($defaultLifetime, $poolHash) {
                 $item = new CacheItem();
                 $item->key = $key;
-                $item->value = $innerItem->get();
+                $item->value = $v = $innerItem->get();
                 $item->isHit = $innerItem->isHit();
                 $item->defaultLifetime = $defaultLifetime;
                 $item->innerItem = $innerItem;
                 $item->poolHash = $poolHash;
+                if (\is_array($v) && 1 === \count($v) && 10 === \strlen($k = \key($v)) && "\x9D" === $k[0] && "\0" === $k[5] && "\x5F" === $k[9]) {
+                    $item->value = $v[$k];
+                    $v = \unpack('Ve/Nc', \substr($k, 1, -1));
+                    $item->metadata[CacheItem::METADATA_EXPIRY] = $v['e'] + CacheItem::METADATA_EXPIRY_OFFSET;
+                    $item->metadata[CacheItem::METADATA_CTIME] = $v['c'];
+                } elseif ($innerItem instanceof CacheItem) {
+                    $item->metadata = $innerItem->metadata;
+                }
                 $innerItem->set(null);
 
                 return $item;
             },
             null,
             CacheItem::class
         );
+        $this->setInnerItem = \Closure::bind(
+            function (CacheItemInterface $innerItem, array $item) {
+                if (isset(($metadata = $item["\0*\0newMetadata"])[CacheItem::METADATA_TAGS])) {
```

Comment:
Same here, Im really trying to understand this, but I can't =)

Reply:
Comment added, better?